cmd/create_test.go

```diff
@@ -9,13 +9,22 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+var templateGeneratedPath string = tests.GetCurrentDirectory()
+
 var _ = BeforeEach(func() {
-	tests.ChangeDirectory(TemplateGeneratedPath + "/test-gin-templates")
+	tests.ChangeDirectory(templateGeneratedPath)
+})
+
+var _ = AfterEach(func() {
+	os.RemoveAll(templateGeneratedPath + "/test-gin-templates")
 })
 
 var _ = Describe("Create template", func() {
 	Context("given the project directory", func() {
 		It("contains project name", func() {
+			cookiecutter := tests.Cookiecutter{AppName: "test-gin-templates"}
+			cookiecutter.CreateProjectFromGinTemplate(templateGeneratedPath)
```

Comment:
I plan to remove `go.sum` and run `go mod tidy` after generated the project from gin-template.
https://github.com/nimblehq/gin-templates/pull/53/files#diff-8601710441b0dde909812d3f5a1d83245fec1f1bf94b6e510d1b72cbd2fb86c1R33

If we re-generate the app for every test, it would take more time than expected as it needs time to download go modules.
Should we keep `go.sum` file or remove it and let `go mod tidy` generate a new one?

Reply:
I think we should keep it so we can get rid of download the go module for every test case. Unit test should isolate the setup step.